test/testTimeSelect.jsx

```diff
@@ -48,66 +44,71 @@ describe('TimeSelect', function() {
+    });
+
+    it('should render the Input field with a value prop', function() {
+      var timeSelect = shallow(<TimeSelect {...props}/>);
+      expect(timeSelect.find(Input).props().value).to.equal(props.value);
     });
 
     it('fills the select box with a range of times', function() {
-      var renderOutput = shallowRender(<TimeSelect />);
-      assert.deepEqual(valuesOfOptions(renderOutput), [
-        '00:30', '01:00', '01:30', '02:00', '02:30', '03:00', '03:30', '04:00',
-        '04:30', '05:00', '05:30', '06:00', '06:30', '07:00', '07:30', '08:00',
-        '08:30', '09:00', '09:30', '10:00', '10:30', '11:00', '11:30', '12:00',
-        '12:30', '13:00', '13:30', '14:00', '14:30', '15:00', '15:30', '16:00',
-        '16:30', '17:00', '17:30', '18:00', '18:30', '19:00', '19:30', '20:00',
-        '20:30', '21:00', '21:30', '22:00', '22:30', '23:00', '23:30'
-      ]);
+      var timeSelect = shallow(<TimeSelect/>);
+      expect(timeSelect.find(Input).children()).to.have.length(47);
     });
 
     it('can adjust the range with start, end and step', function() {
-      var renderOutput = shallowRender(<TimeSelect start={1000} end={1130} step={15} />);
-      assert.deepEqual(valuesOfOptions(renderOutput), [
-        '10:00', '10:15', '10:30', '10:45', '11:00', '11:15'
-      ]);
+      var timeSelect = shallow(<TimeSelect start={1000} end={1130} step={15} />);
+      expect(timeSelect.find(Input).children()).to.have.length(6);
     });
 
     it('will not generate broken times if given strings for start, end and step', function() {
       // Will print warnings in development though, of course
-      var renderOutput = shallowRender(<TimeSelect start="1000" end="1130" step="15" />);
-      assert.deepEqual(valuesOfOptions(renderOutput), [
-        '10:00', '10:15', '10:30', '10:45', '11:00', '11:15'
-      ]);
+      var timeSelect = shallow(<TimeSelect start={'1000'} end={'1130'} step={'15'} />);
+      expect(timeSelect.find(Input).children()).to.have.length(6);
     });
 
     it('can be localised', function() {
-      var renderOutput = shallowRender(<TimeSelect start={1000} end={1130} step={15} locales={['en-US']} />);
-
-      // All options contain a FormattedTime child node from the react-intl library
-      renderOutput.props.children.forEach(function(option) {
-        assert.equal(option.props.children.type, ReactIntl.FormattedTime);
-      });
-
-      // FormattedTimes expect date instances as a starting point to be formatted
-      assert.deepEqual(renderOutput.props.children.map(function(option) {
-        var date = option.props.children.props.value;
-        return [ date.getHours(), date.getMinutes() ];
-      }), [ [10, 0], [10, 15], [10, 30], [10, 45], [11, 0], [11, 15] ]);
+      var timeSelect = shallow(<TimeSelect start={1000} end={1130} step={15} locale="en-US"/>);
+      expect(timeSelect.find(ReactIntl.FormattedTime)).to.have.length(6);
```

Comment:
This doesn't really seem to test the localisation. Is it possible to choose a locale that uses a different time format and test that the values are as expected?